EMI: Keep time of chores and animations in sync

chkr-private opened the pull request:
Both the animation components/tracks and the chores itself keep track of
the current time independently:

animation component:
- AnimationStateEmi::_time
- initialized with 0
- after the first update() call, _time is advanced by the frametime

chores:
- Costume::_currTime
- initialized with -1
- during the first update() call, _currTime is set first to 0
- during the next update() call, _currTime is advanced by the frametime

EMICostume::update() updates first the chore and then all components
with the same frame time. This has the effect, that the internal time
base of the chore and the animation has always an initial offset (the
value of the frametime of the first update() call).

For some costumes, the chore will pause the animation by setting the
according key for that animation at the end of the animation. However,
since the animation component has a different time base, it has already
reached that point and may have already disabled itself entirely
(instead of just being paused by the chore).

This bug fix avoids setting _currTime to 0 in the first update() call of
the chore but just sets it to the frametime. Any consecutive update()
call will advance the time by the frametime. This will keep the two
time bases of the chore and the animation component in sync.

This fixes #1048.

Akz- commented:
I suppose another option would be to change AnimationStateEmi so the time would be 0 during the first update after the animation is started? That would ensure the first animation keyframe is always displayed.

chkr-private commented:
Updated the commit and implemented the bug fix in AnimationStateEmi instead. There is no visible difference. However, probably it is safer to change the animations than the chores.

This PR would change am unsigned value in the savegames to a signed value. So newer savegames would allow negative numbers. Loading a newer savegame with an older engine may cause some trouble, but that can't be prevented. On the other hand, the newer engine can deal with older savegames unless the stored time would be larger than 2 billions. But that's unlikely.

Akz- commented on engines/grim/emi/animationemi.cpp:
```diff
@@ -138,7 +138,7 @@ void AnimationStateEmi::update(uint time) {
 
 	if (!_paused) {
-		uint durationMs = (uint)_anim->_duration;
+		int durationMs = (uint)_anim->_duration;
```
Unnecessary cast here.

Akz- commented:
Looks good to me :+1:

chkr-private commented:
I just updated to commit and corrected the cast.
